applications/nrf_desktop/src/modules/hid_forward.c

```diff
@@ -478,6 +480,66 @@ static void send_nodata_response(const struct config_event *event,
 	EVENT_SUBMIT(rsp);
 }
 
+static void handle_config_channel_peers_req(const struct config_event *event)
+{
+	/* ID 0x00 is reserved for local config channel operations. */
+	static const uint8_t cfg_chan_base_id = 0x01;
+
+	BUILD_ASSERT(ARRAY_SIZE(peripherals) + cfg_chan_base_id <= UCHAR_MAX);
+
+	static uint8_t cur_peer;
+
+	switch (event->status) {
+	case CONFIG_STATUS_INDEX_PEERS:
+		for (size_t i = 0; i < ARRAY_SIZE(peripherals); i++) {
+			if (is_peripheral_connected(&peripherals[i])) {
+				peripherals[i].cfg_chan_id =
+					cfg_chan_base_id + i;
+			}
+		}
+
+		cur_peer = 0;
+		send_nodata_response(event, CONFIG_STATUS_SUCCESS);
+		break;
+
+	case CONFIG_STATUS_GET_PEER:
+		while ((peripherals[cur_peer].cfg_chan_id == CFG_CHAN_UNUSED_PEER_ID) &&
+		       (cur_peer < ARRAY_SIZE(peripherals))) {
+			cur_peer++;
+		}
+
+		struct config_event *rsp = generate_response(event,
+						    HWID_LEN + sizeof(uint8_t));
+		size_t pos = 0;
+
+		if (cur_peer < ARRAY_SIZE(peripherals)) {
+			BUILD_ASSERT(sizeof(peripherals[cur_peer].hwid) ==
+				     HWID_LEN);
+			memcpy(&rsp->dyndata.data[pos],
+			       peripherals[cur_peer].hwid,
+			       sizeof(peripherals[cur_peer].hwid));
+			pos += sizeof(peripherals[cur_peer].hwid);
+
+			rsp->dyndata.data[pos] = peripherals[cur_peer].cfg_chan_id;
+			cur_peer++;
+		} else {
+			memset(&rsp->dyndata.data[pos], 0x00, HWID_LEN);
```

Comment:
we don't need to set anything to hwid part if protocol defines the for CFG_CHAN_UNUSED_PEER_ID it must be ignored

Reply:
Ok, removed. I wanted to use predefined value here to avoid sending "random" data.